storagemarket/impl/providerstates/provider_states.go

```diff
@@ -356,22 +336,6 @@ func FailDeal(ctx fsm.Context, environment ProviderDealEnvironment, deal storage
 
 	log.Warnf("deal %s failed: %s", deal.ProposalCid, deal.Message)
 
-	if !deal.ConnectionClosed {
```

Comment:
So: blocking issue here -- 

This code is used to send a response when the deal is rejected for whatever reason. I believe you still need connection closed for the moment but you need to move it the FSM state change for ProviderEventDataRequested.

And then leave this code here for now

Reply:
The more radical solution I proposed is take the deal validation & decision code out of the statemachine entirely.

Reply:
alternatively, on more possible solution is to add a new state StorageDealSendRejectionResponse and put this code inside of it before moving on to StorageDealFailing. This would enable you to still delete connection closed.